ecommerce_integrations/woocommerce/doctype/woocommerce_setting/woocommerce_setting.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2021, Frappe and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+from frappe.custom.doctype.custom_field.custom_field import create_custom_field
+from frappe.model.document import Document
+from frappe.utils.nestedset import get_root_of
+from six.moves.urllib.parse import urlparse
+
+from ecommerce_integrations.patches.update_woocommerce_items import create_ecommerce_items
+from ecommerce_integrations.woocommerce.constants import PRODUCT_GROUP
+
+
+class WoocommerceSetting(Document):
+	def validate(self):
+		self.validate_settings()
+		self.create_delete_custom_fields()
+		self.create_webhook_url()
+		if self.enable_sync:
+			create_ecommerce_items()
+
+	def create_delete_custom_fields(self):
+		if self.enable_sync:
+			for doctype in ["Customer", "Sales Order", "Item", "Address"]:
```

Comment:
```suggestion
			for doctype in ["Customer", "Sales Order", "Address"]:
```

If e-commerce item is being used then Item should not need a custom field now.